chart/compass/charts/gateway/values.yaml

```diff
@@ -19,5 +19,11 @@ gateway:
     enabled: false
     authMode: "basic"
 
+rateLimit:
+  headerValue: "{{ print .Extra.consumerType }}"
```

Comment:
Shouldn't this also contain the consumerID?  Otherwise, if there is a [noisy neighbor](https://searchcloudcomputing.techtarget.com/definition/noisy-neighbor-cloud-computing-performance), the envoy will rate limit everyone instead of only the one generating load.

Is it possible for envoy to map a token_bucket based on two headers or a regex? This way for each request we can have consumerType and consumerId, then we will have bucket configuration per type and for each consumer of that type, a new bucket will be created. (The values, in this case, will be much smaller of course).

Reply:
Unroturnatly that (token bucket per customer) is not possible using only Istio local rate limiting. The global rate-limiting supports that functionality and yes, that would solve the noisy neighbor problem but in order to use the global rate-limiting, we would need to deploy an additional service that handles all the token buckets for each customer. This is out of scope for this PR.